contracts/staking/Unpack.sol

```diff
@@ -0,0 +1,151 @@
+// (c) 2024, Ava Labs, Inc. All rights reserved.
+// See the file LICENSE for licensing terms.
+// SPDX-License-Identifier: UNLICENSED
+// slither-disable-next-line solc-version
+pragma solidity ^0.8.0; // solhint-disable-line compiler-version
+
+// GENERATED CODE - Do not edit
+
+// solhint-disable no-inline-assembly
+// slither-disable-start assembly
+// Mixed-case can't apply to numbers and an underscore is the most natural separator
+// solhint-disable func-name-mixedcase
+
+library Unpack {
+    /// @dev Thrown if the input buffer to an unpack function is of the wrong length.
+    error IncorrectInputLength(uint256 expected, uint256 actual);
```

Comment:
how strongly do you feel about custom errors? We previously reverted from custom errors back to `require` statements, because we felt some tooling weren't updated to custom errors and lacked in debugging

Reply:
Which tooling lacks support?

I think they're superior to string errors:
1. Don't need to store the entire string in the contract code
2. Can be parameterised to aid debugging and allow precise path detection in tests
3. Avoid [change-detector tests](https://testing.googleblog.com/2015/01/testing-on-toilet-change-detector-tests.html)

Reply:
we reverted to `require` statements over a year ago, so probably need to revisit this. I think back then some combination of `forge` errors, block explorers, and e2e testing just showed the hex error signature. This still seems like the case for our e2e test code, but there's likely a workaround we could do. 

I am also leaning towards custom errors now, but think we should tackle this in a separate issue, and still conform to require to match the rest of the repo.

Reply:
It's not a hill I'll die on, but something I feel I should push back on and ask "are you sure?" before I implement a toggle between the two.

[Independent benchmarking](https://medium.com/frak-defi/solidity-gas-efficiency-unleashed-revert-with-error-vs-revert-with-string-a-deep-dive-248f2cf6bcf2#:~:text=on%20every%20transaction!-,Benchmark%20Comparison,-To%20showcase%20the) of both deployment and usage gas.

Where is the problematic e2e test code?

Reply:
This folder holds the e2e tests I'm referring to https://github.com/ava-labs/teleporter/tree/main/tests/flows. This is also not a hill I'll die on, since I'm also leaning towards custom errors at this point, but wasn't sure to defer to a separate issue and changing all the `require` statements in the repo together so we can keep them consistent for now.